### DISCUSSION TITLE
Pyright can't resolve imports, how to check which python interpreter pyright is using?

### DISCUSSION THREAD

**Discussion by aaronyicongfu** [microsoft/pyright#4202] [General]
Hello, I have run into this issue where pyright simply can't resolve any imports. For example, running
```
echo "import numpy as np" > test_pyright.py && pyright test_pyright.py
```
gives
```
No configuration file found.
No pyproject.toml file found.
stubPath /Users/fyc/toys/python/typings is not a valid directory.
Assuming Python platform Darwin
Searching for source files
Found 1 source file
pyright 1.1.280
/Users/fyc/toys/python/test_pyright.py
  /Users/fyc/toys/python/test_pyright.py:1:8 - error: Import "numpy" could not be resolved (reportMissingImports)
1 error, 0 warnings, 0 informations
Completed in 0.418sec
```

I suspect that pyright is not configured properly to link to the correct interpreter, is there any ways to check this? Thanks in advance for any suggestions.

**Comment by erictraut** [comment]
By default, pyright uses the interpreter that is configured at the time you run pyright. Try typing `python3` in your shell. Which python environment does that start? That's the one pyright will use.

**Reply by aaronyicongfu** [reply]
> By default, pyright uses the interpreter that is configured at the time you run pyright. Try typing `python3` in your shell. Which python environment does that start? That's the one pyright will use.

Problem solved, thanks Eric! I was expecting that pyright would use ```python``` in PATH, which turned out to be different from ```python3``` in my case and caused the issue.

**Reply by danield-catalyst** [reply]
Thank you, this happened to me because brew installed python 3.12 to resolve some dependency and linked python3 so my conda environment was split brained python -> conda, python3 -> brew. A brew unlink python3 solved it for me. Now to track down why brew did that.....
